#163310609 API endpoint for getting all political parties

Raymond-Osy opened the pull request:
#### What does this PR do?
It's an API endpoint for getting all parties

#### Description of Task to be completed?
- Implement feedback for welcome route and response style guide format
- Add parties route to handle an incoming request to this endpoint
- Add parties controller that contains logic for the response
- Add a dummy database to store our data structure

#### How should this be manually tested?
- clone this app to your local computer
- open your console/command line
- type in 'npm run dev'
- open your browser or your postman and on the url type in http://localhost:8000/api/v1/parties/
- a list of all the political parties should be displayed

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
N/A

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

Raymond-Osy committed:
feature(getAllParties): API endpoint for getting all parties
- Implement feedback for welcome route and response style guide
- Add parties route to handle incoming request to this endpoint
- Add parties controller that contains a logic for the response
- Add dummy database to store our data structure

yomigeek commented on server/app.js:
```diff
@@ -1,17 +1,35 @@
+app.all('/*', (req, res) => {
+  res.status(404).send({
+    status: '404',
+    message: 'The page you are looking for was not found'
```
The sentence should be `The page you were looking for was not found!`

Raymond-Osy replied:
Alright, Noted and implemented

yomigeek commented on server/controllers/parties.js:
```diff
@@ -0,0 +1,19 @@
+const app = express();
+app.use(express.json());
+
+class Parties {
```
Your controller file name should be in Pascal casing and singular with controller added to it, this also should apply to the class naming in the file.

Raymond-Osy replied:
Does this apply to all other file names? or just for the controllers?

yomigeek commented on server/controllers/parties.js:
```diff
@@ -0,0 +1,19 @@
+        error: 'No Parties available at this time'
+      });
+    }
+    return res.json({ parties });
```
Please check the documentation for the correct format of the response messages.

yomigeek commented on server/database/dummydbParties.js:
```diff
@@ -0,0 +1,22 @@
+  {
+    id: 1,
+    name: 'APC',
+    hqaddress: '123 williams Akins, Lagos, Nigeria',
```
This should be in camel casing, please correct across the page.

yomigeek commented on server/routes/route.js:
```diff
@@ -0,0 +1,8 @@
+
+router.use('/parties', partyRouter);
+
+export default router;
```
Don't you think it's better you the file name is index.js instead of route.js considering that will be entry file in routes folder?

Raymond-Osy committed:
chore(controllers): feedback to follow conventions in naming
- rename controllers file and class name to pascal namings
- change the entry point for route to index
- follow documentations for response messages
[Finishes #163310609]

yomigeek reviewed (approved):
Merge!
